File: engine/material/material.go
package material

import "github.com/go-gl/mathgl/mgl32"

type Material struct {
	AmbientColor  mgl32.Vec3 `ui:"ColorEdit3"`  // 环境
	DiffuseColor  mgl32.Vec3 `ui:"ColorEdit3"`  // 漫反射
	SpecularColor mgl32.Vec3 `ui:"ColorEdit3"`  // 镜面反射
	Shininess     float32    `ui:"SliderFloat"` // 镜面反射光泽
}

File: engine/material/material.cpp
#include "material.h"
#include "../utils/utils.h"

/*
 * 材质（Material）：描述物体表面对光的响应参数
 *
 * 纯数据类：AmbientColor/DiffuseColor/SpecularColor/Shininess
 * 由 TechniqueLight::SetMaterial() 同步到着色器 gMaterial uniform。
 * 成员默认值定义于 material.h 中。
 */
Material::Material() {
    // 构造时自动生成唯一 UUID
    Id = Utils::GenerateUUID();
}

Material::~Material() {
}